src/mon/MDSMonitor.cc

```diff
@@ -547,6 +547,16 @@ bool MDSMonitor::prepare_beacon(MMDSBeacon *m)
         m->put();
         return false;
       }
+    } else if (state == MDSMap::STATE_DAMAGED) {
+      // Record this MDS rank as damaged, so that other daemons
+      // won't try to run it.
+      pending_mdsmap.up.erase(info.rank);
+      const utime_t until = ceph_clock_now(g_ceph_context);
+      pending_mdsmap.last_failure_osd_epoch = mon->osdmon()->blacklist(
+          info.addr, until);
+      pending_mdsmap.damaged.insert(info.rank);
+      pending_mdsmap.mds_info.erase(gid);  // last! info is a ref into this map
```

Comment:
I'm not sure if I'm missing something about this comment or if it's written wrong? Sure info is a reference into that map...I guess you want the erase to be done last?

Oh, because we can't reference info again after we do this. Yeah, that was confusing, maybe clarify that a bit.